icu4c/source/test/cintltst/ulocbuildertst.c

```diff
@@ -0,0 +1,1743 @@
+    if (U_FAILURE(status) || l2.isBogus()) {
+        errln("build got Error: %s\n", u_errorName(status));
+    }
+
+    if (l1 != l2) {
+        errln("Two locales should be the same, but one is '%s' and the other is '%s'",
+              l1.getName(), l2.getName());
+    }
+}
+
+static void TestPosixCases() {
+    UErrorCode status = U_ZERO_ERROR;
+    Locale l1 = Locale::forLanguageTag("en-US-u-va-posix", status);
+    if (U_FAILURE(status) || l1.isBogus()) {
+        errln("build got Error: %s\n", u_errorName(status));
+    }
+    LocaleBuilder bld;
+    bld.setLanguage("en")
+        .setRegion("MX")
+        .setScript("Arab")
+        .setUnicodeLocaleKeyword("nu", "Thai")
+        .setExtension('x', "1");
+    // All of above should be cleared by the setLocale call.
+    Locale l2 = bld.setLocale(l1).build(status);
+    if (U_FAILURE(status) || l2.isBogus()) {
+        errln("build got Error: %s\n", u_errorName(status));
+    }
+    if (l1 != l2) {
+        errln("The result locale should be the set as the setLocale %s but got %s\n",
+              l1.toLanguageTag<std::string>(status).c_str(),
+              l2.toLanguageTag<std::string>(status).c_str());
+    }
+    Locale posix("en-US-POSIX");
+    if (posix != l2) {
+        errln("The result locale should be the set as %s but got %s\n",
+              posix.getName(), l2.getName());
+    }
+}
+*/
+
+#define TESTCASE(name) addTest(root, &name, "tsutil/ulocbuildertst/" #name)
+void addLocaleBuilderTest(TestNode** root)
+{
+  /*
+    TESTCASE(TestLocaleBuilder);
+    TESTCASE(TestLocaleBuilderBasic);
+    TESTCASE(TestLocaleBuilderBasicWithExtensionsOnDefaultLocale);
+  */
+    TESTCASE(TestSetLanguageWellFormed);
+    TESTCASE(TestSetLanguageIllFormed);
+    TESTCASE(TestSetScriptWellFormed);
+    TESTCASE(TestSetScriptIllFormed);
+    TESTCASE(TestSetRegionWellFormed);
+    TESTCASE(TestSetRegionIllFormed);
+    TESTCASE(TestSetVariantWellFormed);
+    TESTCASE(TestSetVariantIllFormed);
+    TESTCASE(TestSetUnicodeLocaleKeywordWellFormed);
+    TESTCASE(TestSetUnicodeLocaleKeywordIllFormedKey);
+    TESTCASE(TestSetUnicodeLocaleKeywordIllFormedValue);
+  /*
+    TESTCASE(TestAddRemoveUnicodeLocaleAttribute);
+    TESTCASE(TestAddRemoveUnicodeLocaleAttributeWellFormed);
+    */
+    TESTCASE(TestAddUnicodeLocaleAttributeIllFormed);
+    /*
+    TESTCASE(TestSetExtensionU);
+    */
+    TESTCASE(TestSetExtensionValidateUWellFormed);
+    TESTCASE(TestSetExtensionValidateUIllFormed);
+    /*
+    TESTCASE(TestSetExtensionT);
+    TESTCASE(TestSetExtensionValidateTWellFormed);
+    TESTCASE(TestSetExtensionValidateTIllFormed);
+    TESTCASE(TestSetExtensionPU);
+    TESTCASE(TestSetExtensionValidatePUWellFormed);
+    TESTCASE(TestSetExtensionValidatePUIllFormed);
+    TESTCASE(TestSetExtensionOthers);
+    TESTCASE(TestSetExtensionValidateOthersWellFormed);
+    TESTCASE(TestSetExtensionValidateOthersIllFormed);
+    TESTCASE(TestSetLocale);
+    TESTCASE(TestPosixCases);
+    */
```

Comment:
Why are so many of these tests commented out?

Reply:
I am still working on them since the API is not yet approved so this PR (except the header part) is not ready for review yet.

Reply:
ready to review now.